asab/abc/module.py

```diff
@@ -3,7 +3,50 @@
 
 class Module(abc.ABC):
 	"""
-	Abstract module class
+	Abstract class for ASAB modules.
+
+	Modules are registered at the module registry `asab.Application.Modules`, managed by an application object.
+	Module can be loaded by ASAB and typically provides one or more Service objects.
+
+	Every module provides asynchronous methods `initialize()` and `finalize()` that are called when the module is being initialized and finalized.
```

Comment:
initialize is called in the very beginning of the runtime. Finalize in the gentle shut down of the application